lib/HLSL/DxilGenerationPass.cpp

```diff
@@ -315,6 +353,16 @@ class DxilGenerationPass : public ModulePass {
 
     GenerateDxilOperations(M, handleMap);
 
+    if (NotOptimized || m_HasDbgInfo) {
```

Comment:
When would we not want to remove local resouce alloca/load/store/phi? Don't we always want to get rid of them?